### ISSUE TITLE
Documenting a class at other places no longer honors the namespace

### ISSUE THREAD

**Issue by michael-decker** [doxygen/doxygen#8216]
**Describe the bug**
Using the special command "\class myNamespace::myClass" does not place "myClass" documentation in "myNamespace"

**Expected behavior**
I occasionally use macros to generate classes. In order to document the macro-generated class I use the "\class myNamespace::myClass" special command. Doxygen used to place "myClass" in "myNamespace", but no longer does.

**Version**
doxygen 1.8.20; Windows 10, 64-bit

**Comment by albert-github** [comment]
You wrote:
> Doxygen used to place "myClass" in "myNamespace", but no longer does.

in which version did you observe this, correct, behavior?

- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by albert-github** [comment]
When you have a simple file aa.h like:
```
/// \file

namespace MyNs
{
  class MyClass
  {
  };
}


/// \namespace MyNs
/// \brief some nice words about the namespace

/// \class MyNs::MyClass
/// \brief some nice words about the class
```
and a default doxygen configuration settings.

With 1.8.20 you get:

![res_1_8_20](https://user-images.githubusercontent.com/5223533/100626231-d711b300-3325-11eb-8f49-bd95fde14ae5.png)

instead of this you expect (current master: 1.9.0 (63d7b19df11eb55a30e4af8fa492ee464ff353f5)):

![res_master](https://user-images.githubusercontent.com/5223533/100626580-4daeb080-3326-11eb-9e62-1c6ad1a1b53b.png)

Is this what you meant?

Example: [example.tar.gz](https://github.com/doxygen/doxygen/files/5616747/example.tar.gz)

**Comment by michael-decker** [comment]
Yes, that’s it.   Thank You.

Michael Decker
Office: +1-585-269-6125
l3harris.com<mailto:l3harris.com>  / michael.decker@l3harris.com<mailto:michael.decker@l3harris.com>
400 Initiative Drive / Rochester, NY 14624 / USA

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
